### PR TITLE
fix(core): logger shouldLog checking lower log level

### PR DESCRIPTION
**Pull request by Codelax**
Debug = 0
Warning = 2

With level as Warning, `logger.ShouldLog(Debug)` returned true, it should return false
`Debug <= Warning` is true (wrong)
`Debug >= Warning` is false (good)

### CODE REVIEW

File: logger/default_logger.go [hunk 1]
```diff
@@ -51,7 +52,7 @@ func (g *loggerT) Errorf(format string, args ...interface{}) {
 // ShouldLog reports whether verbosity level l is at least the requested verbose level.
 func ShouldLog(level LogLevel) bool { return logger.ShouldLog(level) }
 func (g *loggerT) ShouldLog(level LogLevel) bool {
-	return level <= g.v
+	return level >= g.v
```
Comment (quantumsheep) [inline logger/default_logger.go:55]: This was fine, why the change?

Comment (Codelax) [reply logger/default_logger.go:55]: This wasn't fine

**Review by quantumsheep — APPROVED** [review]
